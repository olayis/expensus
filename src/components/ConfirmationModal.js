import React from 'react';
import Modal from 'react-modal';

const ConfirmationModal = (props) => {
  if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#app');

  return (
    <div>
      <Modal
        contentLabel={props.modalTitle}
        isOpen={props.modalSelected}
        onRequestClose={props.cancelModalSelection}
        closeTimeoutMS={200}
        className='modal'
        aria={{
          labelledby: 'heading',
          describedby: 'full_description',
        }}
      >
        <h3 id='heading' className='modal__title'>
          {props.modalTitle}
        </h3>
        <p id='full_description' className='modal__body'>
          {props.modalDescription}
        </p>
        <button
          className='modal__btn--action'
          onClick={props.handleModalAction}
        >
          {props.actionButtonLabel}
        </button>
        <button
          className='modal__btn--cancel'
          onClick={props.cancelModalSelection}
        >
          Cancel
        </button>
      </Modal>
    </div>
  );
};

export default ConfirmationModal;
