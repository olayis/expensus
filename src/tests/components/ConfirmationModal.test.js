import React from 'react';
import { shallow } from 'enzyme';
import ConfirmationModal from '../../components/ConfirmationModal';

test('should render ConfirmationModal on confirmation button click', () => {
  const wrapper = shallow(
    <ConfirmationModal
      modalSelected={true}
      handleModalAction={jest.fn()}
      cancelModalSelection={jest.fn()}
      modalTitle='MODAL TITLE'
      modalDescription='MODAL DESCRIPTION'
      actionButtonLabel='ACTION'
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should not render ConfirmationModal if action button is not clicked', () => {
  const wrapper = shallow(<ConfirmationModal modalSelected={false} />);
  expect(wrapper).toMatchSnapshot();
});

test('should run action on confirmation', () => {
  const onConfirmation = jest.fn();
  const wrapper = shallow(
    <ConfirmationModal
      modalSelected={true}
      handleModalAction={onConfirmation}
      cancelModalSelection={jest.fn()}
      modalTitle='MODAL TITLE'
      modalDescription='MODAL DESCRIPTION'
      actionButtonLabel='ACTION'
    />
  );
  wrapper.find('.modal__btn--action').simulate('click');
  expect(onConfirmation).toHaveBeenCalled();
});

test('should close confirmationModal and cancel action on cancel button click', () => {
  const cancelModalSelection = jest.fn();
  const wrapper = shallow(
    <ConfirmationModal
      modalSelected={true}
      handleModalAction={jest.fn()}
      cancelModalSelection={cancelModalSelection}
      modalTitle='MODAL TITLE'
      modalDescription='MODAL DESCRIPTION'
      actionButtonLabel='ACTION'
    />
  );
  wrapper.find('.modal__btn--cancel').simulate('click');
  expect(cancelModalSelection).toHaveBeenCalled();
});
