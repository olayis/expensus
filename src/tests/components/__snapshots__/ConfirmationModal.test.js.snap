// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render ConfirmationModal if action button is not clicked 1`] = `
<div>
  <Modal
    aria={
      Object {
        "describedby": "full_description",
        "labelledby": "heading",
      }
    }
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="modal"
    closeTimeoutMS={200}
    isOpen={false}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <h3
      className="modal__title"
      id="heading"
    />
    <p
      className="modal__body"
      id="full_description"
    />
    <button
      className="modal__btn--action"
    />
    <button
      className="modal__btn--cancel"
    >
      Cancel
    </button>
  </Modal>
</div>
`;

exports[`should render ConfirmationModal on confirmation button click 1`] = `
<div>
  <Modal
    aria={
      Object {
        "describedby": "full_description",
        "labelledby": "heading",
      }
    }
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="modal"
    closeTimeoutMS={200}
    contentLabel="MODAL TITLE"
    isOpen={true}
    onRequestClose={[MockFunction]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <h3
      className="modal__title"
      id="heading"
    >
      MODAL TITLE
    </h3>
    <p
      className="modal__body"
      id="full_description"
    >
      MODAL DESCRIPTION
    </p>
    <button
      className="modal__btn--action"
      onClick={[MockFunction]}
    >
      ACTION
    </button>
    <button
      className="modal__btn--cancel"
      onClick={[MockFunction]}
    >
      Cancel
    </button>
  </Modal>
</div>
`;
